flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/reuse/ScanReuserUtils.java

```diff
@@ -203,7 +204,13 @@ private static WatermarkPushDownSpec adjustWatermarkIndex(
                 String name = newFieldNames.get(i);
                 LogicalType type = newSourceType.getTypeAt(i);
                 if (name.equals(rowtimeColumn)) {
-                    type = new TimestampType(type.isNullable(), TimestampKind.ROWTIME, 3);
+                    if (type instanceof LocalZonedTimestampType) {
```

Comment:
Can a Zoned Timestamp type get through to this code? I know this is not what we would expect for a watermark - but wondered if it can get here.